### ISSUE TITLE
There is no option to perform static linking when compiling

### ISSUE THREAD

**Issue by AustinOldham** [godotengine/godot-cpp#461]
In order to make it easier to distribute my project, I compile with the /MT flag instead of /MD. I have to modify the SCons file when I compile but it would be more convenient if I could enable that with an argument. If the change would be accepted, I'm willing to write the patch myself.

**Comment by Calinou** [comment]
Isn't godot-cpp already statically linked by default? I get an `.a` file when compiling on Linux/macOS.

**Comment by petergloor** [comment]
I would love to be able to statically link, at least on Linux but also for Windows if possible.

**Comment by Calinou** [comment]
Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by 2shady4u** [comment]
@Calinou  I ran into the exact same issue.
This issue is **only** present on MSVC and results in a failure to load the dynamic library on `windows`.
![106620010-c6a0d300-6536-11eb-9b41-47698bf9ec4b](https://user-images.githubusercontent.com/42484461/106876885-7c575780-66d8-11eb-9d5b-bc9f1de7d4f3.PNG)

I've had this issue with a user of mine and it was incredibly difficult to figure out.
https://github.com/2shady4u/godot-sqlite/issues/31

Opening the DLL in [DependenciesGUI](https://github.com/lucasg/Dependencies) showcases the fact that indeed the `VCRUNTIME140.DLL` has to be present:
![106875449-f981cd00-66d6-11eb-9194-4dbb12940182](https://user-images.githubusercontent.com/42484461/106876923-84af9280-66d8-11eb-97f1-5a39c52cdf5a.png)

The fix is simply to change the /MD-flag to /MT in the SConstruct file (and /MDd to /MTd for debug mode respectively) 
Or is there a way to automatically install missing VC++ redistributables in Godot?

**Comment by Calinou** [comment]
> Or is there a way to automatically install missing VC++ redistributables in Godot?

No. In fact, you're not even allowed to distribute `vcredist.dll` per the EULA. You need to make your players run the VC++ redistributable installer which requires administrator privileges.
